### ISSUE TITLE
Maximum function nesting level of '250' reached, aborting!

### ISSUE THREAD

**Issue by canfiax** [laravel-ardent/ardent#281]
Having run 3.0.0 successfully, after recently updating using `composer update` I get the following error, when I do:

    $model = My\Models\Model::find(1);

Returns:

    Maximum function nesting level of '250' reached, aborting!

This worked before running composer update.

**Comment by canfiax** [comment]
Funny thing is, if I instead do:

   $model = My\Models\Model::whereId(1)->first();

It works

**Comment by canfiax** [comment]
I have tried switching back to the initial 3.0.0 release, but the error is the same. Guess I haven't used `::find()` for a while.

**Comment by igorsantos07** [comment]
Guess I haven't used find in a while as well, @canfiax.
It seems the issue is explained on #283. See if it's fixed for you in 3.1.1!

**Comment by rmarliere** [comment]
I'm also having this error, but for the version 2.5.0. I had to rollback to version 2.4.2.

**Comment by ManiacTwister** [comment]
@igorsantos07 
I have this problem too. I'm using version 3.3.0 and laravel 5.2  (Is it even supposed to work with 5.2?) 

When i call find() (or all() and probably any other database action) i get memory exhaused because of a infinite loop:

```
LaravelArdent\\Ardent\\Ardent::find() /var/www/app/Http/Controllers/BlogController.php:27, 
LaravelArdent\\Ardent\\Ardent->__call() /var/www/vendor/laravelbook/ardent/src/Ardent/Ardent.php:914, 
Illuminate\\Database\\Eloquent\\Model->__call() /var/www/vendor/laravelbook/ardent/src/Ardent/Ardent.php:387, 
LaravelArdent\\Ardent\\Ardent->newQuery() /var/www/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php:3546, 
LaravelArdent\\Ardent\\Ardent->applyGlobalScopes() /var/www/vendor/laravelbook/ardent/src/Ardent/Ardent.php:934, 
LaravelArdent\\Ardent\\Ardent->__call() /var/www/vendor/laravelbook/ardent/src/Ardent/Ardent.php:934, 
Illuminate\\Database\\Eloquent\\Model->__call() /var/www/vendor/laravelbook/ardent/src/Ardent/Ardent.php:387, 
LaravelArdent\\Ardent\\Ardent->newQuery() /var/www/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php:3546, 
LaravelArdent\\Ardent\\Ardent->applyGlobalScopes() /var/www/vendor/laravelbook/ardent/src/Ardent/Ardent.php:934, 
LaravelArdent\\Ardent\\Ardent->__call() /var/www/vendor/laravelbook/ardent/src/Ardent/Ardent.php:934, 
[...]
```

**Comment by igorsantos07** [comment]
@ManiacTwister, would you provide us with a gist of the affected model, the
table structure, and the exact call you're making? I do use all() and
find() but do not face those issues.

Igor Santos
-- Desenvolvedor Web
[enviado do meu celular]

**Comment by ManiacTwister** [comment]
Here is the gist: https://gist.github.com/ManiacTwister/0dbb10dd67142e7e1000

**Comment by weyforth** [comment]
I just ran into this issue myself. Looks like laravel/framework@ca15fe7317e6cf3a1a138893c804a9c4196904a5 removed `applyGlobalScopes()` and moved it's logic to `newQuery()` causing `_call()` on the Ardent and Eloquent models to loop.

**Comment by shtricul** [comment]
+1 Same problem. Please fix it.

**Comment by igorsantos07** [comment]
@weyforth got it right, the issue is in the snippet by @shtricul. This is a pretty small fix, related to changed in Laravel between 4 and 5. The weirdest thing is it only happens sometimes, when it should happen every time we do any kind of find, I think.

The fix will be published soon on 3.4.2, as this is a bugfix for an internal API. However, PLEASE use caution when updating, I'm not entire sure it would not bring any troubles with it.
